modules/tooling-support-parent/tooling-support/src/main/java/org/mule/runtime/module/tooling/internal/config/InternalDeclarationSession.java

```diff
@@ -121,6 +147,141 @@ public ValueResult getValues(ParameterizedElementDeclaration component, String p
+        if (metadataKeyBuilder == null) {
+          metadataKeyBuilder = MetadataKeyBuilder.newKey(id).withPartName(parameterModel.getName());
+          rootMetadataKeyBuilder = metadataKeyBuilder;
+        } else {
+          MetadataKeyBuilder metadataKeyChildBuilder = MetadataKeyBuilder.newKey(id).withPartName(parameterModel.getName());
+          metadataKeyBuilder.withChild(metadataKeyChildBuilder);
+          metadataKeyBuilder = metadataKeyChildBuilder;
+        }
+      }
+    }
+
+    if (metadataKeyBuilder == null) {
+      return MetadataKeyBuilder.newKey(NullMetadataKey.ID).build();
+    }
+    return rootMetadataKeyBuilder.build();
+  }
+
+  private List<ParameterModel> getMetadataKeyParts(ComponentModel componentModel) {
+    return componentModel.getAllParameterModels().stream()
+        .filter(p -> p.getModelProperty(MetadataKeyPartModelProperty.class).isPresent())
+        .sorted(comparingInt(p -> p.getModelProperty(MetadataKeyPartModelProperty.class).get().getOrder()))
+        .collect(toList());
+  }
+
+  private <T extends ComponentModel> Map<String, Object> getComponentElementDeclarationParameters(ComponentElementDeclaration componentElementDeclaration,
+                                                                                                  T model) {
+    Map<String, Object> parametersMap = new HashMap<>();
+
+    Map<String, ParameterGroupModel> parameterGroups =
+        model.getParameterGroupModels().stream().collect(toMap(NamedObject::getName, identity()));
+
+    List<String> parameterGroupsResolved = new ArrayList<>();
+
+    for (ParameterGroupElementDeclaration parameterGroupElement : componentElementDeclaration.getParameterGroups()) {
+      final String parameterGroupName = parameterGroupElement.getName();
+      final ParameterGroupModel parameterGroupModel = parameterGroups.get(parameterGroupName);
+      if (parameterGroupModel == null) {
+        throw new MuleRuntimeException(createStaticMessage("Could not find parameter group with name: %s in model",
+                                                           parameterGroupName));
+      }
+
+      parameterGroupsResolved.add(parameterGroupName);
+
+      for (ParameterElementDeclaration parameterElement : parameterGroupElement.getParameters()) {
+        final String parameterName = parameterElement.getName();
+        final ParameterModel parameterModel = parameterGroupModel.getParameter(parameterName)
+            .orElseThrow(() -> new MuleRuntimeException(createStaticMessage("Could not find parameter with name: %s in parameter group: %s",
+                                                                            parameterName, parameterGroupName)));
+        parametersMap.put(parameterName,
+                          extractValue(parameterElement.getValue(),
+                                       artifactHelper().getParameterClass(parameterModel, componentElementDeclaration)));
+      }
+    }
+
+    // Default values
```

Comment:
This would be fine for resolving types but when resolving keys with a partial key as input we should not use default values and request to get the key completed in the declaration instead.

Reply:
we could add a flag to `getComponentElementDeclarationParameters` to indicate if defaults should be used or not, that would do?

Reply:
Yes, we could.